elastic/logstash
# deprecate pipeline bus v1

## Release notes

 - Deprecates v1 of the pipeline bus; the v2 implementation has been the default since its introduction in Logstash 8.15.0

## What does this PR do?

Adds a deprecation warning if Logstash is configured to use the legacy v1 pipeline bus. The legacy implementation was replaced with a more-performant one in 8.15.0, and v1 was only left in place as an "escape hatch".

~~The ability to select v1 implementation is set to be removed on `main` in advance of 9.0 via https://github.com/elastic/logstash/pull/16644~~

## Why is it important/What is the impact to the user?

It guides users away from the original lock-heavy implementation, which was previously only available as an escape hatch and is no longer necessary.

## Checklist




## How to test this PR locally

1. add `-Dlogstash.pipelinebus.implementation=v1` to `config/jvm.options`
2. start logstash
3. observe relevant entry in the deprecation log
